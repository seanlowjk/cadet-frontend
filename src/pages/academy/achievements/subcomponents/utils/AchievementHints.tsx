import React from 'react';
import { Tag, Intent } from '@blueprintjs/core';

type AchievementHintsProps = {};

function AchievementHints(props: AchievementHintsProps) {
  const {} = props;

  return (
    <div className="hints">
      <div>
        <Tag round={true} intent={Intent.WARNING}>
          {'NEW'}
        </Tag>
      </div>
      <div>
        <Tag round={true}>{'Week 1'}</Tag>
      </div>
    </div>
  );
}

export default AchievementHints;
