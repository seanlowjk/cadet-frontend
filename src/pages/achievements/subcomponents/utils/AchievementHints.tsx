import React from 'react';
import { Tag, Intent } from '@blueprintjs/core';
import { semester1Weeks } from '../../../../commons/mocks/AchievementMocks';

type AchievementHintsProps = {
  release?: Date;
};

function AchievementHints(props: AchievementHintsProps) {
  const { release } = props;

  const getWeekNumber = (): number | undefined => {
    if (release === undefined) {
      return undefined;
    }

    for (const date in semester1Weeks) {
      if (release <= semester1Weeks[date]) {
        return parseInt(date) - 1;
      }
    }

    return 13;
  };

  return (
    <div className="hints">
      <div>
        {getWeekNumber() === undefined ? (
          <></>
        ) : (
          <Tag intent={Intent.WARNING} round={true}>{`Week ${getWeekNumber()}`}</Tag>
        )}
      </div>
    </div>
  );
}

export default AchievementHints;
