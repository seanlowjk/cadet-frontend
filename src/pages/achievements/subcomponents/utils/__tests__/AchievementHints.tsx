import { mount } from 'enzyme';
import React from 'react';

import AchievementHints from '../AchievementHints';

const mockProps = {
  release: new Date()
};

test('AchievementExp component renders correctly', () => {
  const app = <AchievementHints {...mockProps} />;
  const tree = mount(app);
  expect(tree.debug()).toMatchSnapshot();
});
