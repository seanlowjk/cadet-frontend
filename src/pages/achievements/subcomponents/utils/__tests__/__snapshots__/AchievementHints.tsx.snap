// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AchievementExp component renders correctly 1`] = `
"<AchievementWeek release={{...}}>
  <div className=\\"week\\">
    <div />
  </div>
</AchievementWeek>"
`;

exports[`AchievementExp component renders correctly 2`] = `
"<AchievementWeek release={{...}}>
  <div className=\\"week\\">
    <div />
  </div>
</AchievementWeek>"
`;

exports[`AchievementExp component renders correctly 3`] = `
"<AchievementWeek release={{...}}>
  <div className=\\"week\\">
    <div />
  </div>
</AchievementWeek>"
`;

exports[`AchievementExp component renders correctly 4`] = `
"<AchievementWeek release={{...}}>
  <div className=\\"week\\">
    <div />
  </div>
</AchievementWeek>"
`;

exports[`AchievementExp component renders correctly 5`] = `
"<AchievementWeek release={{...}}>
  <div className=\\"week\\">
    <div />
  </div>
</AchievementWeek>"
`;

exports[`AchievementExp component renders correctly 6`] = `
"<AchievementWeek release={{...}}>
  <div className=\\"week\\">
    <div />
  </div>
</AchievementWeek>"
`;
